### PR TITLE
ci: split Analytics stage

### PR DESCRIPTION
**Pull request by skaldarnar**
Splits the one big 'Analytics' stage of tests, static code analysis, and
javadoc generation into smaller stages with dedicated purpose.

- **Unit Tests** run unit tests and upload report. this is not expected
to fail, thus using default health scale factor of 1.0
- **Integration Tests** run the heavy-weight MTE tests (and alike) and
upload report. we know they are flaky, so keep them after the publish
stage, and keep the health scale factor of 0.0
- **Analytics** run static code checks (spotbugs, PMD, ...) and upload
reports
- **Documentation** generate javadoc and upload result. this could
potentially be the home for a static site generation step in the future

Contributes to https://github.com/MovingBlocks/Terasology/issues/4607

Co-authored-by: Josephine Rueckert <jd.rueckert@googlemail.com>

### CODE REVIEW

**Comment by jdrueckert** [comment]
[Separate Jenkins test stages](https://trello.com/c/fgorgiQY/53-separate-jenkins-test-stages)

File: Jenkinsfile [hunk 1]
```diff
@@ -41,21 +46,26 @@ node ("ts-module && heavy && java8") {
...
     }
 
+    stage('Integration Tests') {
+        sh './gradlew integrationTest'
+        junit testResults: 'build/test-results/integrationTest/*.xml', allowEmptyResults: true, healthScaleFactor: 0.0
+    }
+
     stage('Analytics') {
-        sh './gradlew check spotbugsmain javadoc'
+        sh './gradlew check -x test spotbugsmain'
```
Comment (skaldarnar) [inline Jenkinsfile:55]:
```suggestion
        sh './gradlew check --exclude-task test spotbugsmain'
```

Comment (Cervator) [reply Jenkinsfile:55]: Would it be a thought to just drop the `check` task entirely and just run `gradlew spotbugsmain pmd checkstyle` or whichever the right task names are? Those could potentially be put into a parallel block later as well as individual steps.

If we eventually add new analytics we'd just have to add another block with the new task. Not sure if `check` really contains anything we wouldn't want to call out explicitly. Less mystery that way.

Will try to test this PR shortly.

Comment (skaldarnar) [reply Jenkinsfile:55]: hm, isn't there this weird split about `spotBugsMain` and `spotBugTests` for everything, and there is no general `spotbugs` task to just run both? :thinking:

Comment (Cervator) [reply Jenkinsfile:55]: Erg, yeah that may be. Although maybe that's just two tasks then. In any case that's a trivial thing we can worry about later if we want to :-)

Works fine!

**Review by Cervator — APPROVED** [review]
Tests out fine 👍 

image

I mixed this in with the JTE v2 upgrade which trivially edits the one other file. Going to check on one more related topic there then merge everything together.

**Pull request merged**